projection/ibc_channel/ibc_channel.go

```diff
@@ -286,35 +286,31 @@ func (projection *IBCChannel) HandleEvents(height int64, events []event_entity.E
 				return fmt.Errorf("error updating channel last_activity_time: %w", err)
 			}
 
-			amount := msgIBCRecvPacket.Params.MaybeFungibleTokenPacketData.Amount.Uint64()
-			denom := msgIBCRecvPacket.Params.MaybeFungibleTokenPacketData.Denom
-			destinationChannelID := msgIBCRecvPacket.Params.Packet.DestinationChannel
-			destinationPortID := msgIBCRecvPacket.Params.Packet.DestinationPort
-
-			if err := projection.updateBondedTokensWhenMsgIBCRecvPacket(
-				ibcChannelsView,
-				channelID,
-				amount,
-				denom,
-				destinationChannelID,
-				destinationPortID,
-			); err != nil {
-				return fmt.Errorf("error updateChannelBondedTokensWhenMsgIBCRecvPacket: %w", err)
+			if msgIBCRecvPacket.Params.PacketAck.MaybeError == "" {
```

Comment:
Similar here

Reply:
This one will be kept, as the `success inverted value bug` in recv packet is not fixed yet.
I added some TODOs in the code base to reminder ourselves to update this part when the bug is fixed 😂 .

Reply:
After the following changes I propose, here should check if the value `== nil`